src/server/storage_engine/meta_engine.rs

```diff
@@ -137,45 +159,44 @@ impl MetaEngine {
                 }
             }
         }
+
+        let mut offset = offset;
         match self.dir_db.db.get(path.as_bytes())? {
-            Some(value) => match bincode::deserialize::<SubDirectory>(&value) {
-                Ok(sub_dir) => {
-                    let iter = sub_dir.sub_dir.iter().skip(offset as usize);
-                    let mut result = Vec::with_capacity(size as usize);
-                    let mut total = 0;
-                    for value in iter {
-                        let r#type = match value.1.as_bytes()[0] {
-                            b'f' => DT_REG,
-                            b'd' => DT_DIR,
-                            b's' => DT_LNK,
-                            _ => DT_REG,
-                        };
-                        let reclen = value.0.len() + 3;
-                        if total + reclen > size as usize {
-                            break;
-                        }
-                        result.extend_from_slice(&r#type.to_le_bytes());
-                        result.extend_from_slice(&(value.0.len() as u16).to_le_bytes());
-                        result.extend_from_slice(value.0.as_bytes());
-                        total += reclen;
+            Some(value) => {
+                let mut result = Vec::with_capacity(size as usize);
+                let mut l = usize::from_le_bytes(value.as_slice().try_into().unwrap());
+                let mut total = 0;
+                for item in self.dir_db.db.iterator(IteratorMode::From(
+                    format!("{}-", path).as_bytes(),
+                    rocksdb::Direction::Forward,
+                )) {
+                    if offset > 0 {
+                        offset -= 1;
```

Comment:
Should add `l -= 1`.

Reply:
But I think the database trait should implement `seek` operation and move this code in it.

Reply:
> But I think the database should implement `seek` operation and move this code in it.

Rocksdb doesn't implement this function

Reply:
> But I think the database should implement `seek` operation and move this code in it.

I don't find that rocksdb implements this function.

Reply:
> Should add `l -= 1`.

Done